src/connection_string.ts

```diff
@@ -551,6 +551,7 @@ export function parseOptions(
   );
 
   mongoOptions.metadata = makeClientMetadata(mongoOptions);
+  mongoOptions.extendedMetadata = addContainerMetadata(mongoOptions.metadata);
```

Comment:
we might want to add a `catch` block, like Anna [did when parallelizing dns lookup and srv resolution](https://github.com/mongodb/node-mongodb-native/pull/4012), so that this doesn't result in an unhandled promise rejection on the off change it rejects

Reply:
Good catch.

Why are we calling this up here again? seems like `addContainerMetadata` has what it needs down in prepareHandshake no?

Reply:
We cache the promise during options parsing so that we only construct the extended metadata once.  Then we just `await` it during every call to connect.  Aditi filed https://jira.mongodb.org/browse/NODE-5994 to make `clientMetadata` private, remove `extendedMetadata` and just store its result in `metadata` in driver v7

Reply:
Ah ok, thanks for the context sorry I missed that.

Reply:
Added in the catch block.